### PR TITLE
add annotation to secrets field

### PR DESCRIPTION
**Pull request by zqingqing1**
**What this PR does / why we need it**:  Followed up on https://github.com/Azure/acs-engine/pull/3227, the CloneAndConform function will added at RP codebase.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ type PublicKey struct {
 type CustomSearchDomain struct {
 	Name          string `json:"name,omitempty"`
 	RealmUser     string `json:"realmUser,omitempty"`
-	RealmPassword string `json:"realmPassword,omitempty"`
+	RealmPassword string `json:"realmPassword,omitempty" `
```
Comment (weinong) [inline pkg/api/types.go]: extra space is not needed

File: pkg/api/types.go [hunk 2]
```diff
@@ -151,15 +151,15 @@ type CustomSearchDomain struct {
 	RealmPassword string `json:"realmPassword,omitempty"`
 }
 
-// CustomNodesDNS represents the Search Domain when the custom vnet for a custom DNS as a nameserver.
+// CustomNodesDNS represents the Search Domain whenthe custom vnet for a custom DNS as a nameserver.
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: why did you remove a space here?

Comment (CecileRobertMichon) [reply pkg/api/types.go]: (admittedly that comment isn't very correct grammatically)
